lib/components/molecules/DateSelector/DateSelector.js

```diff
@@ -0,0 +1,184 @@
+// @flow
+/**
+ *
+ * DateSelector
+ *
+ */
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import classNames from 'classnames';
+import styles from './DateSelector.style';
+import type { Props } from './types';
+import Select from '../../atoms/Select';
+import Label from '../../atoms/Label';
+
+const DateSelector = ({
+  id,
+  className,
+  format,
+  startDate,
+  endDate,
+  locale,
+  dateLabel,
+  monthLabel,
+  yearLabel,
+}: Props): Node => {
+  const [selectedDate, setSelectedDate] = useState(new Date());
+  const [months, setMonths] = useState([]);
+
+  const getMonths = () => {
+    const date = new Date();
+    const monthsOptions = [];
+    for (let i = 0; i < 12; i += 1) {
+      date.setMonth(i);
+      monthsOptions.push(date.toLocaleString(locale, { month: 'long' }));
+    }
+    return monthsOptions;
+  };
+
+  useEffect(() => {
+    setMonths(getMonths());
+  }, []);
+
+  useEffect(() => {
+    setSelectedDate(startDate);
+  }, [startDate, endDate]);
+
+  const getDaysInMonth = () => {
+    const daysOptions = [];
+    let startDay = 1;
+    let endDay = new Date(selectedDate.getFullYear(), selectedDate.getMonth() + 1, 0).getDate();
+    if (
+      selectedDate.getMonth() === startDate.getMonth() &&
+      selectedDate.getFullYear() === startDate.getFullYear()
+    ) {
+      startDay = startDate.getDate();
+    }
+    if (
+      selectedDate.getMonth() === endDate.getMonth() &&
+      selectedDate.getFullYear() === endDate.getFullYear()
+    ) {
+      endDay = endDate.getDate();
+    }
+    for (let i = startDay; i <= endDay; i += 1) {
+      daysOptions.push(i);
+    }
+    return daysOptions;
+  };
+
+  const getMonthOptions = () => {
+    let monthOptions = [...months];
+    if (selectedDate.getFullYear() === startDate.getFullYear()) {
+      const startMonth = startDate.getMonth();
+      monthOptions = monthOptions.slice(startMonth, months.length);
+    }
+    if (selectedDate.getFullYear() === endDate.getFullYear()) {
+      const endMonth = endDate.getMonth();
+      monthOptions = monthOptions.slice(0, endMonth + 1);
+    }
+    return monthOptions;
+  };
+
+  const getYearOptions = () => {
+    const years = [];
+    for (let i = startDate.getFullYear(); i <= endDate.getFullYear(); i += 1) {
```

Comment:
Can't we generate this ES6 way?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from

https://stackoverflow.com/questions/39924644/es6-generate-an-array-of-numbers